openstack/tools/charmed_openstack_functest_runner.sh

```diff
@@ -148,13 +160,36 @@ if $MODIFY_BUNDLE_CONSTRAINTS; then
     )
 fi
 
-for target in ${func_targets[@]}; do
+for target in ${!func_targets[@]}; do
     [[ -d src ]] && pushd src || true
     tox -re func-target -- $target
 
+    if (($?==0)); then
+        func_targets[target]='success'
+    else:
+        func_targets[target]='fail'
+    fi
+
     read -p "Destroy model and run next test? [ENTER]"
     # cleanup before next run
     model=`juju list-models| egrep -o "^zaza-\S+"|tr -d '*'`
     juju destroy-model --no-prompt $model --force --no-wait --destroy-storage
 done
 
+# Report results
+echo "Test results for charm $CHARM_NAME functional tests @ commit $COMMIT_ID:"
+for target in ${!func_targets[@]}; do
+    if $(python3 $TOOLS_PATH/test_is_voting.py) $target; then
```

Comment:
Why not `if python3 $TOOLS_PATH/test_is_voting.py $target; then`?

Reply:
yep that was a bug, fixed now